internal/cmd/config/add.go

```diff
@@ -0,0 +1,101 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"slices"
+
+	"github.com/spf13/cobra"
+
+	"github.com/hetznercloud/cli/internal/cmd/cmpl"
+	"github.com/hetznercloud/cli/internal/cmd/util"
+	"github.com/hetznercloud/cli/internal/state"
+	"github.com/hetznercloud/cli/internal/state/config"
+)
+
+func NewAddCommand(s state.State) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "add <key> <value>...",
+		Short:                 "Add values to a list",
+		Long:                  "Add values to a list. For a list of all available configuration options, run 'hcloud help config'.",
+		Args:                  util.Validate,
+		TraverseChildren:      true,
+		DisableFlagsInUseLine: true,
+		SilenceUsage:          true,
+		RunE:                  state.Wrap(s, runAdd),
+		ValidArgsFunction: cmpl.NoFileCompletion(cmpl.SuggestArgs(
+			cmpl.SuggestCandidatesF(func() []string {
+				var keys []string
+				for key, opt := range config.Options {
+					if opt.IsSlice() && opt.HasFlag(config.OptionFlagPreference) {
+						keys = append(keys, key)
+					}
+				}
+				return keys
+			}),
+			cmpl.SuggestCandidatesCtx(func(_ *cobra.Command, args []string) []string {
+				var comps []string
+				if opt, ok := config.Options[args[0]]; ok {
+					comps = opt.Completions()
+				}
+				return comps
+			}),
+		)),
+	}
+	cmd.Flags().Bool("global", false, "Set the value globally (for all contexts)")
+	return cmd
+}
+
+func runAdd(s state.State, cmd *cobra.Command, args []string) error {
+	global, _ := cmd.Flags().GetBool("global")
+
+	var (
+		added []any
+		ctx   config.Context
+		prefs config.Preferences
+	)
+
+	if global {
+		prefs = s.Config().Preferences()
+	} else {
+		ctx = s.Config().ActiveContext()
+		if util.IsNil(ctx) {
+			return fmt.Errorf("no active context (use --global flag to set a global option)")
+		}
+		prefs = ctx.Preferences()
+	}
```

Comment:
I saw this appear multiple times in the code, might be a candidate for a dedicated function.